### ISSUE TITLE
16 Bit Training Experiences

### ISSUE THREAD

**Issue by mbreuss** [columbia-ai-robotics/scalingup#17]
Hi,

thank you for sharing this well organized codebase!

I was reading the extended documentation and noticed a comment about training issues with 16 bit.
I am using 16 bit training for my current diffusion policy setup, without any issues (At least that I am aware of:D)
Could you elaborate on these problems you encountered? Was it about training loss instabilities or something else?

Thanks in advance:)

**Comment by huy-ha** [comment]
Hey! Yea, I always got some training loss spike when using 16 bit a few epochs into training, and the network usually doesn't recover afterwards. It should be very obvious if it doesn't work. Did you just run the commands in the README but change float to half? Also which GPU are you using?

**Comment by mbreuss** [comment]
Okay, thats interesting. I reimplemented the model into my own project with different training data, where I am using Pytorch Lightning included 16 bit precision training and never experienced any issues. I tested the training on RTX 3090s and on A6000s.

**Comment by huy-ha** [comment]
This codebase also uses Pytorch Lightning, and I tested on 3090s and A6000s too 🥲 When I tried half, I also use Lightning trainer's precision argument... Did you change much about my implementation or did you use the original diffusion policy's implementation?

**Comment by mbreuss** [comment]
Oh thats weird. So the model consists of the U-net code from the diffusion policy repo combined with two ResNets-18, also from Diffusion-Policy. For language encoding I used the CLIP with OpenAI weights. Are there some major differences in your implementation from the other repo, that are important for your performance?  The only major difference is, that I used the continous-time diffusion objective with it. However, I cannot image, that the ddpm objective would be less stable, since I use a wide noise range from 0.001 to 80.

**Comment by huy-ha** [comment]
There shouldn't be any algorithmic or many hyperparameter differences between mine and the other implementation (I started from that implementation, and refactored it). I don't think I will look into this discrepancy further, but you do end up doing it, I would really appreciate you letting the community know! Thanks Moritz.

**Comment by mbreuss** [comment]
Okay, thats good to know. If I encounter anything suspicious  I will share it. Thanks for all the answers!
